sdk/src/client/secret/mod.rs

```diff
@@ -115,6 +116,160 @@ pub trait SecretManage: Send + Sync {
     ) -> Result<TransactionPayload, Self::Error>;
 }
 
+#[async_trait]
+pub trait DynSecretManage: std::fmt::Debug + Send + Sync {
+    async fn dyn_generate_ed25519_addresses(
+        &self,
+        coin_type: u32,
+        account_index: u32,
+        address_indexes: Range<u32>,
+        options: Option<GenerateAddressOptions>,
+    ) -> Result<Vec<Ed25519Address>, Error>;
+
+    async fn dyn_generate_evm_addresses(
```

Comment:
Not sure if it would be more annoying, but should we maybe split the evm things into an own trait? Many people are probably not interested at that and would then still need to implement this with an error or something

Reply:
The main issue with that is that it means more bounds and in the end we'll probably have to require `T: Ed25519SecretManage + EvmSecretManage` most places anyway so they'd still have to.

Reply:
Hm but where do we need `EvmSecretManage`? Not in the client and wallet, only on the secret managers (at least currently)

Reply:
Well, the problem is we can only have one dynamic secret manager, so it's either only an `dyn Ed25519SecretManage` or `dyn Ed25519SecretManage + EvmSecretManage`, both have their downsides. For secret manager methods, I doubt we even need to use the dynamic stuff really, we can just use a generic. If there's never a use for the evm fns outside of the secret manager methods, then splitting it might make sense.